core/templates/pages/oppia-root/app.module.ts

```diff
@@ -95,6 +96,7 @@ export class MyHammerConfig extends HammerGestureConfig {
     AppRoutingModule,
     I18nModule,
     ToastrModule.forRoot(toastrConfig),
+    SharedComponentsModule,
```

Comment:
Why do we need this here? Don't we load this in the other modules?

Reply:
This is to solve the issue of the unsaved changes modal not appearing when migrating. It will also help prevent modal related issues in other migrations like exploration editor and blog dashboard.

Reply:
@mon4our Could you please answer why we don't need this in the other modules? Why is this PR "special" in this regard?

Reply:
@seanlip We do need this in other modules as well. The only reason of importing it here is for certain modals like UnsavedChangesStatusInfoModalComponent (skill editor), ExplorationSaveModalComponent (exploration editor), etc to work as expected and not throw 'No component factory found for...' and other modal related errors.

I've also asked Srijan about this, waiting for his response. PTAL, thanks!

Reply:
@mon4our But what exactly is causing the issue? Is there another way to include it in the necessary modules rather than including it here?

Here's why I'm concerned: as I understand it, this app.module gets imported on all pages. The shared components and shared services stuff are temporary things that are supposed to go away after the angular migration. As you can see, the SharedComponents module file basically includes almost every component in the app. It does not seem like we are going in the right direction if you are putting this here, since this entails loading all of these components, including the unnecessary ones, in every Oppia page. So I think it is important to find a more fine-grained way to include the necessary modules in the skill editor page and not just put them here.

Reply:
Ok, got it. The issue is the `UnsavedChangesStatusInfoModalComponent` is opened by the `skill-editor-staleness-detection.service.ts` which has 
```
@Injectable({
  providedIn: 'root',
})
```
and since the `app.module.ts` is loaded first and `skill-editor-page.module.ts` is lazy loaded, `app.module.ts` becomes the 'root' and gets searched for the component factory for `UnsavedChangesStatusInfoModalComponent`, giving the 'No component factory found for...' error (That is why importing SharedComponentsModule in `app.module.ts` here was fixing the issue). To prevent this we can provide `SkillEditorStalenessDetectionService` in the providers of `skill-editor-page.module.ts` which fixes the issue!

Reply:
I see the issue is already resolved. Thats the correct solution. This modules seem like a lot of mess but soon we will able to get of all the modules in our codebase :)